omnipaxos/src/sequence_paxos/follower.rs

```diff
@@ -144,15 +134,38 @@ where
                         ],
                         "storage error while trying to write snapshot",
                     );
-                    let accepted_idx = self.internal_storage.append_entries(accsync.suffix);
+                    let accepted_res = self
+                        .internal_storage
+                        .append_entries_without_batching(accsync.suffix);
+                    // manually rollback set_snapshot if append suffix fails
+                    if let Err(_e) = &accepted_res {
+                        self.internal_storage
+                            .try_trim(self.internal_storage.get_accepted_idx())
+                            .expect("storage error while trying to trim log entries");
+                        self.internal_storage
+                            .append_entries_without_batching(old_log)
+                            .expect("storage error while trying to rollback log entries");
```

Comment:
If I understand correctly, we first trim because we essentially want to clear the log before appending the old entries.  To make this a bit less messy/avoid confusion I think both the trimming and then appending the old log should be hidden from here. I propose we create a new function called `rollback_log()` in InternalStorage.